Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm

```diff
@@ -4369,6 +4402,214 @@ HTTPServer server({
+{
+    static constexpr auto mainFrameBytes = R"TESTRESOURCE(
+    <div id='text'>mainframe content</div>
+    <script>
+        function getTextContent() {
+            window.webkit.messageHandlers.testHandler.postMessage(document.getElementById('text').innerHTML);
+        }
+        function getIframeTextContent() {
+            document.getElementById('iframe').contentWindow.postMessage('print', '*');
+        }
+        function postResult(event) {
+            window.webkit.messageHandlers.testHandler.postMessage(event.data);
+        }
+        addEventListener('message', postResult, false);
+    </script>
+    <iframe id='iframe' src='https://apple.com/apple'></iframe>
+    )TESTRESOURCE"_s;
+
+    static constexpr auto iframeBytes = R"TESTRESOURCE(
+    <div id='text'>iframe content</div>
+    <script>
+        addEventListener('message', () => {
+            let textElement = document.getElementById('text');
+            parent.postMessage(textElement.innerHTML, '*');
+        }, false);
+        parent.postMessage('loaded', '*');
+    </script>
+    )TESTRESOURCE"_s;
+
+    HTTPServer server({
+        { "/example"_s, { mainFrameBytes } },
+        { "/apple"_s, { iframeBytes } },
+    }, HTTPServer::Protocol::HttpsProxy);
+
+    bool receivedMessage1 = false;
+    bool receivedMessage2 = false;
+    bool receivedMessage3 = false;
+    auto webViewAndDelegates = makeWebViewAndDelegates(server);
+    [webViewAndDelegates.messageHandler addMessage:@"loaded" withHandler:[&]() {
+        receivedMessage1 = true;
+    }];
+    [webViewAndDelegates.messageHandler addMessage:@"MAINFRAME CONTENT" withHandler:[&]() {
+        receivedMessage2 = true;
+    }];
+    [webViewAndDelegates.messageHandler addMessage:@"IFRAME CONTENT" withHandler:[&]() {
+        receivedMessage3 = true;
+    }];
+    RetainPtr webView = webViewAndDelegates.webView;
+    RetainPtr textManipulationDelegate = adoptNS([[SiteIsolationTextManipulationDelegate alloc] init]);
+    [webView _setTextManipulationDelegate:textManipulationDelegate.get()];
+    RetainPtr manipulationConfiguration = adoptNS([[_WKTextManipulationConfiguration alloc] init]);
+    manipulationConfiguration.get().includeSubframes = YES;
+
+    [webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"https://example.com/example"]]];
+    Util::run(&receivedMessage1);
+
+    [webView _startTextManipulationsWithConfiguration:manipulationConfiguration.get() completion:^{ }];
+    while ([textManipulationDelegate items].count < 2)
+        Util::spinRunLoop();
+
+    RetainPtr items = [textManipulationDelegate items];
+    auto sortFunction = ^(_WKTextManipulationItem *item1, _WKTextManipulationItem *item2) {
+        auto value1 = [NSNumber numberWithBool:item1.isSubframe];
+        auto value2 = [NSNumber numberWithBool:item2.isSubframe];
+        return [value1 compare:value2];
+    };
+    RetainPtr sortedItems = [items.get() sortedArrayUsingComparator:sortFunction];
```

Comment:
Why do we need to sort the results? Given we're waiting for both the main frame and the iframe to load, the order should be deterministic: prefix DFS.

Reply:
The items are received from different web processes, so there is no guarantee they will arrive UI process in order.
(i.e. UI process send "start" to WP1, WP2, WP3 in order and the replies may not come in WP1, WP2, WP3 order)

Reply:
Hm... maybe we should check with the translation team to make sure that's okay. Right now, iframe content comes in the order they appear. Changing the order of it could affect translation results.